### PR TITLE
Move tests to new test-app

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
V2 addons don't have an integrated test setup atm, so a separate test app is how we test v2 addons.

Test-helpers isn't a v2 addon yet, but this'll greatly reduce the diff of that upcoming change.

This is a non-breaking change as it mostly focuses on using a new app for the tests instead of using the "dummy app"

-----------

For V2 Conversion
- extract test-app <- this PR
- fix type-imports 
- find old code and remove it (optional, unless it's tied to the build)
- native library conversion

depending on how these PRs go will determine if we need to do a major or not.

### CODE REVIEW

File: test-app/tests/test-helper.js [hunk 1]
```diff
@@ -35,19 +38,6 @@ QUnit.done(function () {
   }
 });
 
-class TestLoader extends AbstractTestLoader {
```
Comment (NullVoxPopuli) [inline test-app/tests/test-helper.js:38]: handled by ember-qunit these days

File: test-app/tests/test-helper.js [hunk 2]
```diff
@@ -109,8 +99,8 @@ QUnit.assert.deprecationsInclude = function (expected) {
   });
 };
 
-QUnit.start();
-
-if (typeof Testem !== 'undefined') {
```
Comment (NullVoxPopuli) [inline test-app/tests/test-helper.js:114]: also handled by ember-qunit

File: test-app/tests/unit/setup-context-test.js [hunk 3]
```diff
@@ -29,12 +29,14 @@ import config from '../../config/environment';
 import Ember from 'ember';
 import { deprecate, warn } from '@ember/debug';
 
+console.log('1', Ember.testing);
```
Comment (lifeart) [inline test-app/tests/unit/setup-context-test.js:32]: seems console logs could be omitted here

File: test-app/package.json [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    "qunit": "^2.21.1",
+    "qunit-dom": "^3.2.0",
+    "tracked-built-ins": "^3.1.1",
+    "webpack": "^5.78.0"
+  },
+  "engines": {
+    "node": "14.* || 16.* || >= 18"
+  },
+  "ember": {
+    "edition": "octane"
```
Comment (lifeart) [inline test-app/package.json:78]: is edition still a thing here?

File: test-app/package.json [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    "loader.js": "^4.7.0",
+    "pretender": "^3.4.7",
+    "prettier": "^2.8.7",
+    "qunit": "^2.21.1",
+    "qunit-dom": "^3.2.0",
+    "tracked-built-ins": "^3.1.1",
+    "webpack": "^5.78.0"
+  },
+  "engines": {
+    "node": "14.* || 16.* || >= 18"
```
Comment (lifeart) [inline test-app/package.json:75]: wondering if we could drop 14 and 16..

File: test-app/package.json (hunk 4 above)
Comment (NullVoxPopuli) [reply test-app/package.json:78]: Yup

File: test-app/tests/unit/setup-context-test.js (hunk 3 above)
Comment (NullVoxPopuli) [reply test-app/tests/unit/setup-context-test.js:32]: Oops. Thanks!

File: test-app/package.json (hunk 5 above)
Comment (NullVoxPopuli) [reply test-app/package.json:75]: We could, but would me separate. During v2 conversion, node would cease to matter anyway

**Review by kategengler — APPROVED** [review]
The Readme needs to be updated to reflect this change (right now says `ember test`) as does CONTRIBUTING.md (or removed, there isn't much there that isn't in the Readme)

**Pull request merged**